README.md

```diff
@@ -149,12 +149,15 @@ The paths passed to the linters are absolute to avoid confusion in case they're
 }
 ```
 
-**NOTE:** Prior to `lint-staged@5`, when the `gitDir` option was present, file patterns used for filtering via minimatch had to be relative to the git root.
-But that's no longer necessary and the `gitDir` option is deprecated. Instead, `lint-staged` will now do the following:
+When mathcing, lint-staged will do the following
 
 * Resolve the git root automatically, no configuration needed.
 * Pick the staged files which are present inside the project directory.
 * Filter them using the specified glob patterns.
+* Resolve filtered paths relative to the git root and make them absolute.
```

Comment:
@okonet I took a through look and I want to drop this line.

For one thing, this isn't accurate. Here's the steps we follow in `generateTasks.js`:

* Extract the staged files (paths are relative).
* Resolve absolute path for each file relative to git root.
* Check that the file is inside the project folder (`process.cwd()`).
* Determine the relative path for each file relative to the project root.
* Filter using minimatch.
* Convert back into absolute path. git root is not involved in this step.

So the statement `Resolve filtered paths relative to the git root and make them absolute.` is not correct. Even if we drop this line, the overall explanation is still good enough IMHO.